controllers/dataprotection/backup_controller.go

```diff
@@ -726,18 +931,20 @@ func (r *BackupReconciler) createUpdatesJobs(reqCtx intctrlutil.RequestCtx,
 
 func (r *BackupReconciler) createMetadataCollectionJob(reqCtx intctrlutil.RequestCtx,
 	backup *dataprotectionv1alpha1.Backup,
+	commonPolicy *dataprotectionv1alpha1.CommonBackupPolicy,
 	basePolicy *dataprotectionv1alpha1.BasePolicy,
+	pathPrefix string,
 	updateInfo dataprotectionv1alpha1.BackupStatusUpdate) error {
-	mgrNS := viper.GetString(constant.CfgKeyCtrlrMgrNS)
+	jobNamespace := viper.GetString(constant.CfgKeyCtrlrMgrNS)
+	if commonPolicy != nil && strings.TrimSpace(updateInfo.Script) == "" {
```

Comment:
this conditions are somewhat obscure and need to be explicitly a marker to use the namespace corresponding to backup, for example using annotations?